kafka-connect-s3/src/main/java/io/confluent/connect/s3/TopicPartitionWriter.java

```diff
@@ -599,7 +607,11 @@ private void commitFiles() {
       String encodedPartition = entry.getKey();
       commitFile(encodedPartition);
       if (isTaggingEnabled) {
-        tagFile(encodedPartition, entry.getValue());
+        RetryUtil.exponentialBackoffRetry(() -> tagFile(encodedPartition, entry.getValue()),
```

Comment:
maybe a better place is to add the retry in `storage.addTags(s3ObjectPath, tags)` (line no 658). That we can have retries prior to the ignore of fail check which should help with the robustness